server/build_event_protocol/build_event_handler/build_event_handler.go

```diff
@@ -782,7 +783,10 @@ func (e *EventChannel) handleEvent(event *pepb.PublishBuildToolEventStreamReques
 	} else if !e.hasReceivedEventWithOptions {
 		e.eventsBeforeOptions = append(e.eventsBeforeOptions, invocationEvent)
 		if len(e.eventsBeforeOptions) > 100 {
-			log.Warningf("We got over 100 build events before an event with options for invocation %s, dropping %+v", iid, e.eventsBeforeOptions[0])
+			if !e.hasDroppedEventsBeforeOptions {
+				e.hasDroppedEventsBeforeOptions = true
+				log.Warningf("We got over 100 build events before an event with options for invocation %s. Beginning to drop events. Earliest dropped event: %+v", iid, e.eventsBeforeOptions[0])
+			}
```

Comment:
thinking about this a little more -- is there a way to log the index of the last event we had to drop?

It'd be nice to know basically, how big of a buffer we'd need to keep in memory to not drop anything.

Reply:
Done